crates/testlang/outputs/cargo/slang_testlang/src/generated/napi_interface/cst.rs

```diff
@@ -2,8 +2,7 @@
 
 use std::rc::Rc;
 
-use napi::bindgen_prelude::Env;
-use napi::JsObject;
+use napi::Either;
```

Comment:
regarding this:

> But having an ability to refer to cst.Node by the users is, I feel, more important than our convenience here.

Just adding context:

We re-export these types from specific TypeScript modules that add this namespace (`npm/package/src/cst/index.ts`) .. So technically we can get rid of custom namespaces in Rust and the NAPI-generated `index.d.ts`, and still provide the namespacing ourselves, since we hide/re-export the symbols from `index.d.ts`.

But then if we get rid of custom namespacing, then we can run into name conflicts if two modules have a common type name (errors, results, etc). Additionally, I was planning on consolidating the Rust vs NAPI types in #570 by roughly doing the following:

```rust
struct Cursor { .... }

#[napi(js_name = "Cursor")]
struct NapiCursor(Cursor);
```

Not sure if that will pick up the correct names if we get rid of `ts_return_type` .. Will need to double check.

Reply:
Additionally, I was planning on consolidating the Rust vs NAPI types in #570 by roughly doing the following:

```rust
struct Cursor { .... }

#[napi(js_name = "Cursor")]
struct NapiCursor(Cursor);
```

Not sure if that will pick up the correct names if we get rid of `ts_return_type` .. Will need to double check.

Reply:
>  So technically we can get rid of custom namespaces (...) and still provide the namespacing ourselves (...)

I'm afraid this trades one explicit step that has to be synchronized (`ts_return_type` and the actual Rust type) with another (we would have to make sure to manually mirror the Rust module structure, we still can get clashing definitions as you mentioned), so unfortunately I don't think that's better over what we have, as much as I'd like to improve how things are right now.

To be clear, since `cst.Node` was brought up in the quote, I want to clarify that the issue of 1. not adding a `type Node = RuleNode | TokenNode` in index.d.ts ourselves and 2. getting rid of `ts_return_type` are slightly orthogonal: we can still remove our definition but use `ts_return_type = Array<cst.RuleNode | cst.TokenNode>`.

Since we seem to agree that `ts_return_type` is here to stay due to the namespace benefits, we could get rid of our additional `cst.Node` definition; it's just that I decided not to due to DX improvement as mentioned in the OP; it's such a common type that having it in a dedicated type alias helps with readability, IMO. I'm up for not having additional, manual definitions as well, though.

Reply:
> Additionally, I was planning on consolidating the Rust vs NAPI types (...)

@OmarTawfik Since I accidentally started dealing a lot more with NAPI and its inner workings recently I can also take that over, if you'd like. To be very explicit, I don't mean to steal work or silo the knowledge but since I'm already deep here, I can take this on to reduce context switching if you feel this would offload you or be beneficial in general.

Reply:
> it's just that I decided not to due to DX improvement as mentioned in the OP

Agreed.

> Since I accidentally started dealing a lot more with NAPI and its inner workings recently I can also take that over, if you'd like

Let's talk about it f2f. Happy to share some of it. I'm mainly working on (90% done) with the codegen infra changes to enable this. I have not actually updated any of the public APIs yet.